diff --git a/makefile b/makefile
index 4cddd35..a278ef8 100644
--- a/makefile
+++ b/makefile
@@ -39,6 +39,9 @@ all: progs
 	
 clean: clean_progs
 
+libs:
+	@$(foreach prog,$(PROJECT_PROGRAM_LOCAL_DIR), cd "$(prog)" && $(MAKE) deps static_lib && cd "$(PROJECT_PATH)";)
+
 # Programs
 .PHONY: progs
 progs:
@@ -46,4 +49,4 @@ progs:
 
 .PHONY: clean_progs
 clean_progs:
-	@$(foreach prog,$(PROJECT_PROGRAM_LOCAL_DIR), cd "$(prog)" && $(MAKE) clean_deps clean && cd "$(PROJECT_PATH)";)
\ No newline at end of file
+	@$(foreach prog,$(PROJECT_PROGRAM_LOCAL_DIR), cd "$(prog)" && $(MAKE) clean_deps clean && cd "$(PROJECT_PATH)";)
