diff --git a/webview.h b/webview.h
index bc09c68..4e03f95 100644
--- a/webview.h
+++ b/webview.h
@@ -2703,7 +2703,8 @@ auto parse_version(const T (&version)[Length]) noexcept {
   return parse_version(std::basic_string<T>(version, Length));
 }
 
-std::wstring get_file_version_string(const std::wstring &file_path) noexcept {
+WEBVIEW_API std::wstring
+get_file_version_string(const std::wstring &file_path) noexcept {
   DWORD dummy_handle; // Unused
   DWORD info_buffer_length =
       GetFileVersionInfoSizeW(file_path.c_str(), &dummy_handle);
@@ -4239,7 +4240,7 @@ using webview = browser_engine;
 
 namespace detail {
 
-webview *cast_to_webview(void *w) {
+WEBVIEW_API webview *cast_to_webview(void *w) {
   if (!w) {
     throw exception{WEBVIEW_ERROR_INVALID_ARGUMENT,
                     "Cannot cast null pointer to webview instance"};
